dev-docs/source/ISISReflectometryInterface.rst

```diff
@@ -0,0 +1,309 @@
+    };
+
+- The presenter implements the subscriber interface.
+
+  .. code-block:: c++
+
+    class MANTIDQT_ISISREFLECTOMETRY_DLL EventPresenter
+        : public IEventPresenter,
+          public EventViewSubscriber
+
+- It overrides the notification functions to perform the relevant actions.
+
+  .. code-block:: c++
+
+    void EventPresenter::notifyUniformSliceCountChanged(int) {
+      setUniformSlicingByNumberOfSlicesFromView();
+      m_mainPresenter->notifySettingsChanged();
+    }
+
+- When a user interacts with the view, all the view needs to do is send the appropriate notification. The view does not know anything about the concrete type that it is notifying. This helps to avoid accidentally introducing logic into the view about what should happen on an event and instead just notify that an event happened. It could also be easily extended to support multiple notifyees of different types. 
+
+  .. code-block:: c++
+
+    void QtEventView::onUniformEvenChanged(int numberOfSlices) {
+      m_notifyee->notifyUniformSliceCountChanged(numberOfSlices);
+    }
+
+
+Dependency injection
+####################
+
+A simple example of dependency injection is in the use of an :code:`IMessageHandler` interface to provide a service to display messages to the user. These messages must be displayed by a Qt view. Rather than each view having to implement this, we use one object (in this case the :code:`QtMainWindowView`) to implement this functionality and pass it as an :code:`IMessageHandler` to all of the presenters that need it.
+
+- The :code:`IMessageHandler` interface defines the functions for displaying messages:
+
+  .. code-block:: c++
+
+    class IMessageHandler {
+    public:
+      virtual void giveUserCritical(const std::string &prompt,
+                                    const std::string &title) = 0;
+      ...
+    };
+
+- The :code:`QtMainWindowView` implements these:
+
+  .. code-block:: c++
+
+    void QtMainWindowView::giveUserCritical(const std::string &prompt,
+                                            const std::string &title) {
+      QMessageBox::critical(this, QString::fromStdString(title),
+                            QString::fromStdString(prompt), QMessageBox::Ok,
+                            QMessageBox::Ok);
+    }
+
+- The :code:`QtMainWindowView` sets up the concrete instance (actually just a pointer to itself) and passes it in the construction of anything that needs it, e.g. the :code:`RunsPresenter` (in this case using a factory - more about the `Factory pattern`_ below):
+
+  .. code-block:: c++
+
+    auto messageHandler = this;
+    auto makeRunsPresenter = RunsPresenterFactory(..., messageHandler);
+
+- The :code:`RunsPresenter` then has a simple service it can use to display messages without needing to know anything about the :code:`QtMainWindowView`:
+
+  .. code-block:: c++
+
+    m_messageHandler->giveUserInfo("Search field is empty", "Search Issue");
+
```

Comment:
I'd also add an example about how we could test that a notification will be sent to Qt if we were in a known critical situation, as it really hammers home the point about testability